test/specs/lib/felaSanitizeCssPlugin-test.ts

```diff
@@ -0,0 +1,84 @@
+import sanitizeCss from 'src/lib/felaSanitizeCssPlugin'
+
+const validStringStyle = {
+  display: 'block',
+  margin: '0 0 0 0',
+}
+
+const validNumericStyle = {
+  top: 0,
+}
+
+const invalidCssValue = 'rgba('
+
+const invalidCssStyle = {
+  color: invalidCssValue,
+}
+
+const assertCssPropertyValue = (value: string, isValid: boolean) => {
+  test(`assert that '${value}' is ${isValid ? 'valid' : 'invalid'}`, () => {
+    const sanitize = sanitizeCss()
+
+    const style = { display: value }
+    const sanitizedStyle = sanitize(style)
+
+    expect(sanitizedStyle).toEqual(isValid ? style : {})
+  })
+}
+
+const sanitize = sanitizeCss()
+
+describe('tests', () => {
+  test('should ensure there are no non-closed brackets in CSS property value', () => {
+    const style = {
+      ...validStringStyle,
+      ...invalidCssStyle,
+    }
+
+    expect(sanitize(style)).toEqual(validStringStyle)
+  })
+
+  test('should skip numeric CSS property values', () => {
+    expect(sanitize(validNumericStyle)).toEqual(validNumericStyle)
+  })
+
+  test('should recursively process nested objects', () => {
+    const style = {
+      ...validStringStyle,
+      '::before': {
+        ...validStringStyle,
+        ...invalidCssStyle,
+      },
+    }
+
+    expect(sanitize(style)).toEqual({
+      ...validStringStyle,
+      '::before': {
+        ...validStringStyle,
+      },
```

Comment:
To the last point, I think having simple objects here would help understanding.  It is not clear what the test is doing from looking at it.  I need to scroll back and forth, memorize two object shapes, and then picture two different merged shapes in my mind to see the assertion.

I think having the objects in simple plain form is more ideal for testing purposes.

Feedback?

Reply:
sure, will do this way, agree